projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-publicize-share-post.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Publicize: Share post
+ *
+ * This file is synced from the Jetpack monorepo to WPCOM.
+ *
+ * @package automattic/jetpack
+ * @since
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+require_once __DIR__ . '/publicize-connections.php';
+
+/**
+ * Publicize: Share post class.
+ */
+class WPCOM_REST_API_V2_Endpoint_Publicize_Share_Post extends WP_REST_Controller {
+	/**
+	 * The constructor sets the route namespace, rest_base, and registers our API route and endpoint.
+	 * Additionally, we check if we're executing this file on WPCOM or Jetpack.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+
+		// $wpcom_is_wpcom_only_endpoint = true keeps WPCOM from trying to loop back to the Jetpack endpoint.
+		$this->wpcom_is_wpcom_only_endpoint = true;
+
+		// Determine if this endpoint is running on WPCOM or not.
+		$this->is_wpcom = false;
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$this->is_wpcom = true;
+		}
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * This file is synced from Jetpack to WPCOM and this method creates a slightly different route for both sites.
+	 * Jetpack route: http://{$site}/wp-json/wpcom/v2/posts/{$postId}/publicize
+	 * WPCOM route: https://public-api.wordpress.com/wpcom/v2/sites/{$siteId}/posts/{$postId}/publicize
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/posts/(?P<postId>\d+)/publicize',
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'share_post' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+				'args'                => array(
+					'message'             => array(
+						'description'       => __( 'The message to share.', 'jetpack' ),
+						'type'              => 'string',
+						'required'          => true,
+						'validate_callback' => function ( $param ) {
+							return is_string( $param ) && '' !== $param;
+						},
+						'sanitize_callback' => 'sanitize_textarea_field',
+					),
+					'skipped_connections' => array(
+						'description'       => __( 'Array of external connection IDs to skip sharing.', 'jetpack' ),
+						'type'              => 'array',
+						'required'          => false,
+						'validate_callback' => function ( $param ) {
+							return is_array( $param );
+						},
+						'sanitize_callback' => function ( $param ) {
+							return array_map( 'absint', $param );
+						},
+					),
+				),
+			),
+			// override = true because this API route was commandeered from the file
+			// wp-content/rest-api-plugins/endpoints/sites-publicize.php on WPCOM.
```

Comment:
Does this mean that the route already exists on WordPress.com, and we'll want this new version to overwrite it once that file is merged here and committed to WordPress.com?

Reply:
> how will this handle shared connections, created and attached to one user, but shared with other users on the site?

AFAIK, it gets the connections hitting the `connection-test-results` endpoint that is not tied to the user but to the site. It means you can configure different connections per site.
Publicize picks the site connections and then creates and stores a new _connections list_ into the post metadata. 
So in theory, all users that have access to the site will see the same connections list.

Reply:
Ok, cool, thanks for the extra details!